pkg/server/structlogging/hot_ranges_log.go

```diff
@@ -103,57 +130,122 @@ func (s *hotRangesLoggingScheduler) startJob(ctx context.Context, stopper *stop.
 }
 
 func (s *hotRangesLoggingScheduler) start(ctx context.Context, stopper *stop.Stopper) error {
-	intervalChangedChan := make(chan struct{})
-	// We have to register this callback first. Otherwise we may run into
-	// an unlikely but possible scenario where we've started the ticker,
-	// and the setting is changed before we register the callback and the
-	// ticker will not be reset to the new value.
-	TelemetryHotRangesStatsInterval.SetOnChange(&s.st.SV, func(ctx context.Context) {
-		intervalChangedChan <- struct{}{}
-	})
-
-	ticker := time.NewTicker(TelemetryHotRangesStatsInterval.Get(&s.st.SV))
-
 	for {
+		checkInterval := CheckInterval
```

Comment:
was this intended to be in the loop?

wouldn't `checkInterval` keep increasing?

Reply:
This is probably a naming thing, but the interval won't increase in the multitenant case, because capital `CheckInterval` never changes (and therefore lowercase `checkInterval` always starts as the value of CheckInterval).

Reply:
ack that makes sense to me, yeah i def got confused with the naming.

Though it feels like we're contorting ourselves in this function to make the tests work. Not sure if there's a cleaner way to get the same test coverage.

Reply:
I thought about this a bit and my instinct is to say that mocking the status server would be the simplest way - though it seems like for server behavior we generally avoid mocking any behavior.

Beyond that, I see other ways to contort the tests to achieve the same functionality, but they seem similarly obtuse (examples include manually altering the warming speed of the replicas, stubbing the log spy where `LogEventsWithDelay` is used).

Lmk if you wanna try either of those, or another approach  - I'm open to trying out alternatives and seeing what they look like.